src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1214,6 +1213,24 @@ public override void VisitMarkupElement(MarkupElementSyntax node)
                 base.VisitMarkupElement(node);
 
                 _builder.Pop();
+
+                static bool LooksLikeAComponentName(DocumentIntermediateNode document, string startTagName)
+                {
+                    var category = CharUnicodeInfo.GetUnicodeCategory(startTagName, 0);
```

Comment:
You need to use `char.GetUnicodeCategory` instead of `CharUnicodeInfo.GetUnicodeCategory`